app/shared/Shared/Models/ChatRequest.cs

```diff
@@ -10,4 +10,7 @@ public record class ChatRequest(
     ) : ApproachRequest(Approach.RetrieveThenRead)
 {
     public string? LastUserQuestion => History?.Last(m => m.Role == "user")?.Content;
+    
+    [JsonPropertyName("connectionId")]
```

Comment:
Is it weird to put connectionId here. It makes clients think it need to pass connectionId but actually you assign it in server side. I suggest not adding connectionId in the protocol between clients and servers.